api/tests/routes/backoffice/offers_test.py

```diff
@@ -908,3 +909,140 @@ def test_batch_reject_offers(self, legit_user, authenticated_client):
+            assert "Lieu : Le Petit Rintintin" in card_text
+
+
+class GetHistoryTest(GetEndpointHelper):
+    endpoint = "backoffice_web.offer.get_history"
+    endpoint_kwargs = {"offer_id": 1}
+    needed_permission = perm_models.Permissions.READ_OFFERS
+
+    def test_get_history_with_no_history(self, legit_user, authenticated_client):
+        offer = offers_factories.OfferFactory()
+
+        history_url = url_for(self.endpoint, offer_id=offer.id, _external=True)
+        history_response = authenticated_client.get(history_url)
+
+        history_rows = html_parser.extract_table_rows(history_response.data)
+        assert len(history_rows) == 0
+
+    def test_get_history_with_validation_offer_history(self, legit_user, authenticated_client):
+        validation_date = datetime.datetime.utcnow()
+        offer = offers_factories.OfferFactory(
+            lastValidationDate=validation_date, lastValidationAuthor=users_factories.UserFactory()
+        )
+
+        form_url = url_for(self.endpoint, offer_id=offer.id, _external=True)
+
+        with assert_num_queries(4):  # session + 2 x user + offer
+            response = authenticated_client.get(form_url)
+            assert response.status_code == 200
+
+            history_rows = html_parser.extract_table_rows(response.data)
+            assert len(history_rows) == 1
+            history_row = history_rows[0]
+            assert history_row["Auteur"] == "Jean NEIGE"
+            assert history_row["Date/Heure"]
+
+    def test_get_history_with_reject_offer_history(self, legit_user, authenticated_client):
+        validation_date = datetime.datetime.utcnow()
+        offer = offers_factories.OfferFactory(
+            lastValidationDate=validation_date,
+            validation=offers_models.OfferValidationStatus.REJECTED,
+            lastValidationAuthor=users_factories.UserFactory(),
+        )
+
+        form_url = url_for(self.endpoint, offer_id=offer.id, _external=True)
+
+        with assert_num_queries(4):  # session + 2 x user + offer
+            response = authenticated_client.get(form_url)
+            assert response.status_code == 200
+
+            history_rows = html_parser.extract_table_rows(response.data)
+            assert len(history_rows) == 1
+            history_row = history_rows[0]
+            assert history_row["Auteur"] == "Jean NEIGE"
+            assert history_row["Date/Heure"]
+
+
+class GetStockTest(GetEndpointHelper):
+    endpoint = "backoffice_web.offer.get_stocks"
+    endpoint_kwargs = {"offer_id": 1}
+    needed_permission = perm_models.Permissions.READ_OFFERS
+
+    def test_get_stocks_with_no_stocks(self, legit_user, authenticated_client):
+        offer = offers_factories.OfferFactory()
+
+        stocks_url = url_for(self.endpoint, offer_id=offer.id, _external=True)
+        stocks_response = authenticated_client.get(stocks_url)
+
+        stocks_rows = html_parser.extract_table_rows(stocks_response.data)
+        assert len(stocks_rows) == 0
```

Comment:
Suggestion de simplification : 
```suggestion
        stocks_rows = html_parser.count_table_rows(stocks_response.data) == 0
```